app/assets/javascripts/modules/bucket-name.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+moj.Modules.bucketName = {
+  inputName: 'new-datasource-name',
+
+  init: function() {
+    var self = this;
```

Comment:
My rough rule of thumb with `this` is that if it's used inside a block, or if it's used more than once, copy it to `self`, so that it's clear what's being referred to. It's true that this works without `self`, but given the prevalence of the pattern, I think it's clearer.

I'm not wedded to it though.

Reply:
Yeah, I got that. It's no big deal.